packages/payment/src/services/payment-module.ts

```diff
@@ -207,38 +207,33 @@ export default class PaymentModuleService<
     data: CreatePaymentSessionDTO,
     @MedusaContext() sharedContext?: Context
   ): Promise<PaymentSessionDTO> {
-    const created = await this.paymentSessionService_.create(
-      {
+    const { result, errors } = await paymentSessionWorkflow(
+      this.__container__ as any
+    ).run({
+      input: {
+        payment_collection_id: paymentCollectionId,
         provider_id: data.provider_id,
-        amount: data.providerContext.amount,
-        currency_code: data.providerContext.currency_code,
-        payment_collection: paymentCollectionId,
+        amount: data.amount,
+        currency_code: data.currency_code,
+        context: data.context,
+        data: data.data,
       },
-      sharedContext
-    )
+      context: sharedContext,
+    })
 
-    try {
-      const sessionData = await this.paymentProviderService_.createSession(
-        data.provider_id,
-        {
-          ...data.providerContext,
-          resource_id: created.id,
-        }
-      )
+    if (errors.length) {
+      throw errors[0].error
+    }
 
-      await this.paymentSessionService_.update(
-        {
-          id: created.id,
-          data: sessionData,
-        },
-        sharedContext
-      )
+    const paymentSession = await this.paymentSessionService_.retrieve(
```

Comment:
comment: The result from the workflow execution cannot be directly serialized because it's missing some required properties. Perhaps, this is expected but thought I'd highlight, since this means we need to introduce some sort of "pattern" here. 

![CleanShot 2024-02-29 at 20 28 31](https://github.com/medusajs/medusa/assets/59018053/c56e82bd-b676-4f64-a7bf-e43e33da8ffc)